File: src/main/kotlin/coraythan/keyswap/patreon/PatreonRewardsTier.kt
package coraythan.keyswap.patreon

enum class PatreonRewardsTier(val tierIds: List<String>, val maxNotifications: Int = 0, val maxApiRequests: Int = coraythan.keyswap.publicapis.maxApiRequests) {
    NOTICE_BARGAINS(listOf("3351357", "5546595")), // $3
    SUPPORT_SOPHISTICATION(listOf("3281536", "5546600"), 25, 50), // $6
    MERCHANT_AEMBERMAKER(listOf("3509683", "5546606"), 25, 100), // $12
    ALWAYS_GENEROUS(listOf("3358580", "3281558"), 50, 250) // $25+
}

fun PatreonRewardsTier?.levelAtLeast(tier: PatreonRewardsTier): Boolean {
    if (this == null) return false
    return this.ordinal >= tier.ordinal
}

File: ui/src/patreon/PatreonRewardsTier.ts
export enum PatreonRewardsTier {
    NOTICE_BARGAINS = "NOTICE_BARGAINS", // $1
    SUPPORT_SOPHISTICATION = "SUPPORT_SOPHISTICATION", // $5
    MERCHANT_AEMBERMAKER = "MERCHANT_AEMBERMAKER", // $10
    ALWAYS_GENEROUS = "ALWAYS_GENEROUS" // $25 Exclusive tier or $50
}

export const findPatronRewardLevel = (tier?: PatreonRewardsTier) => {
    switch (tier) {
        case PatreonRewardsTier.NOTICE_BARGAINS:
            return 1
        case PatreonRewardsTier.SUPPORT_SOPHISTICATION:
            return 2
        case PatreonRewardsTier.MERCHANT_AEMBERMAKER:
            return 3
        case PatreonRewardsTier.ALWAYS_GENEROUS:
            return 4
        default:
            return 0
    }
}

export const patronRewardLevelName =  (tier?: PatreonRewardsTier) => {
    switch (tier) {
        case PatreonRewardsTier.NOTICE_BARGAINS:
            return "It that Fastidiously Notices Bargains"
        case PatreonRewardsTier.SUPPORT_SOPHISTICATION:
            return "They who Artfully Support Sophistication"
        case PatreonRewardsTier.MERCHANT_AEMBERMAKER:
            return "Merchant Æmbermaker"
        case PatreonRewardsTier.ALWAYS_GENEROUS:
            return "Butterfield the Always Generous"
        default:
            return ""
    }
}
